tgui/packages/tgui/interfaces/GasCanister/Detonator.js

```diff
@@ -0,0 +1,210 @@
+import { Fragment } from 'inferno';
+import { Button, LabeledList, Section, Box, Flex, Divider } from '../../components';
+import { DetonatorTimer } from './DetonatorTimer';
+
+export const Detonator = props => {
+  const {
+    detonator,
+    detonatorAttachments,
+    onToggleAnchor,
+    onToggleSafety,
+    onWireInteract,
+    onPrimeDetonator,
+    onTriggerActivate,
+    onSetTimer,
+  } = props;
+
+  return (
+    <Section
+      title="Detonator">
+      <DetonatorWires
+        detonator={detonator}
+        onWireInteract={onWireInteract}
+        onSetTimer={onSetTimer} />
+      <Divider />
+      <DetonatorUtility
+        detonator={detonator}
+        detonatorAttachments={detonatorAttachments}
+        onToggleAnchor={onToggleAnchor}
+        onToggleSafety={onToggleSafety}
+        onPrimeDetonator={onPrimeDetonator}
+        onTriggerActivate={onTriggerActivate} />
+    </Section>
+  );
+};
+
+const DetonatorWires = props => {
+  const {
+    detonator: {
+      wireNames,
+      wireStatus,
+      time,
+      isPrimed,
+    },
```

Comment:
Minor, but if you're destructuring a nested object within props, make sure it has an empty object as a default so it doesn't fall over trying to pull values off of `undefined`.
```suggestion
    } = {},
```

Reply:
done